lightningd/gossip/gossip.c

```diff
@@ -105,6 +105,47 @@ static struct peer *setup_new_peer(struct daemon *daemon, const u8 *msg)
 	return peer;
 }
 
+static struct io_plan *owner_msg_in(struct io_conn *conn,
+				    struct daemon_conn *dc);
+static struct io_plan *nonlocal_dump_gossip(struct io_conn *conn,
+					    struct daemon_conn *dc);
+
+/* When a peer is to be owned by another daemon, we create a socket
+ * pair to send/receive gossip from it */
+static void send_peer_with_fds(struct peer *peer, const u8 *msg)
```

Comment:
Oh, you already moved the passing of the owner socket fd to the release phase. This should cut down on fds in the main daemon if all we get are gossiping peers.